repo.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"sync"
+)
+
+type repository interface {
+	Get(id int) (event, bool)
+	GetAll() []event
+	Set(e event) error
+	Delete(id int) error
+}
+
+func NewInMemRepo() *InMemRepo {
+	return &InMemRepo{events: make([]event, 0)}
+}
+
+type InMemRepo struct {
+	sync.Mutex
+	events []event
+}
+
+func (imr *InMemRepo) Get(id int) (event, bool) {
```

Comment:
Returning a bool here was a kinda hot take. I just wanted to do `if _, ok := {...}; !ok {...}` stuff. An error would probably be better since that makes the interface more flexible for stuff which can error - like writing to a DB or writing to disk